backend/fms_core/services/sample.py

```diff
@@ -429,6 +429,120 @@ def pool_samples(process: Process,
+        container_destination: a container object that will receive the pool.
+        coordinates_destination: the coordinate on the container where the pool is stored.
+        reception_date: The date which the pool was received .
+
+    Returns:
+        a tuple containing the following data.
+        sample_destination: a sample object that represents the new pool.
+        errors: errors that were encountered during the pool creation.
+        warnings: warnings to inform the user of potential mistakes or dangerous operations.
+    """
+    pool_sample_obj = None
+    errors = []
+    warnings = []
+
+    if not samples_info:
+        errors.append(f"Information must be provided about the samples to pool.")
+    if not container_destination:
+        errors.append(f"Destination container for extraction is required.")
+    if not pool_name:
+        errors.append(f"Pool Name is required.")
+    if not isinstance(reception_date, date):
+        errors.append(f"Reception date is not valid.")
+
+    if not errors:
+        # Calculate the total volume of the pool
+        pool_volume = sum(sample["volume"] for sample in samples_info)
+        for sample in samples_info:
+            sample_volume = sample["volume"]
+            # Calculate the volume ratio of each sample in the pool
+            sample["volume_ratio"] = decimal_rounded_to_precision(sample_volume / pool_volume)
+        # Create a a new sample - Concentration value is not set (need a QC to set it)
+        pool_data = dict(
+            name=pool_name,
+            volume=pool_volume,
+            creation_date=reception_date,
+            container=container_destination,
+            **(dict(comment=comment) if comment is not None else f"Automatically generated by pooling submission on {datetime.utcnow().isoformat()}Z"),
+            **(dict(coordinates=coordinates_destination) if coordinates_destination is not None else dict()),
+        )
+        try:
+            pool_sample_obj = Sample.objects.create(**pool_data)
+        except Exception as e:
+            errors.append(e)
+
+        if pool_sample_obj:
+            for sample in samples_info:
+                volume_ratio = sample["volume_ratio"]
+
+                # Create Derived Samples and Biosamples for each sample
+                biosample_data = dict(
+                    collection_site=sample['collection_site'],
+                    **(dict(individual=sample['individual']) if sample['individual'] is not None else dict()),
+                    **(dict(alias=sample['alias']) if sample['alias'] is not None else dict()),
+                )
+
+                try:
+                    biosample = Biosample.objects.create(**biosample_data)
+
+                    derived_sample_data = dict(
+                        biosample_id=biosample.id,
+                        sample_kind=sample['sample_kind'],
+                        **(dict(library=sample['library']) if sample['library'] is not None else dict()),
```

Comment:
Do not allow empty library to be part of a pool. This would be an issue since we do not have an integrity test for that. Make sure to test and set an error if library is None­.